src/Extensions/AzureBlobPayloads/Converters/LargePayloadDataConverter.cs

```diff
@@ -0,0 +1,93 @@
+/// <exception cref="ArgumentNullException">Thrown when <paramref name="innerConverter"/>, <paramref name="payloadStore"/>, or <paramref name="largePayloadStorageOptions"/> is null.</exception>
+public sealed class LargePayloadDataConverter(DataConverter innerConverter, IPayloadStore payloadStore, LargePayloadStorageOptions largePayloadStorageOptions) : DataConverter
+{
+    const string TokenPrefix = "blob:v1:"; // matches BlobExternalPayloadStore
+
+    readonly DataConverter innerConverter = innerConverter ?? throw new ArgumentNullException(nameof(innerConverter));
+    readonly IPayloadStore payLoadStore = payloadStore ?? throw new ArgumentNullException(nameof(payloadStore));
+    readonly LargePayloadStorageOptions largePayloadStorageOptions = largePayloadStorageOptions ?? throw new ArgumentNullException(nameof(largePayloadStorageOptions));
+    // Use UTF-8 without a BOM (encoderShouldEmitUTF8Identifier=false). JSON in UTF-8 should not include a
+    // byte order mark per RFC 8259, and omitting it avoids hidden extra bytes that could skew the
+    // externalization threshold calculation and prevents interop issues with strict JSON parsers.
+    // A few legacy tools rely on a BOM for encoding detection, but modern JSON tooling assumes BOM-less UTF-8.
+    readonly Encoding utf8 = new UTF8Encoding(false);
+
+    /// <summary>
+    /// Serializes the value to a JSON string and uploads it to the external payload store if it exceeds the configured threshold.
+    /// </summary>
+    /// <param name="value">The value to serialize.</param>
+    /// <returns>The serialized value or the token if externalized.</returns>
+    public override string? Serialize(object? value)
+    {
+        if (value is null)
+        {
+            return null;
+        }
+
+        string json = this.innerConverter.Serialize(value) ?? "null";
+
+        int byteCount = this.utf8.GetByteCount(json);
+        if (byteCount < this.largePayloadStorageOptions.ExternalizeThresholdBytes)
+        {
+            return json;
+        }
+
+        // Upload synchronously in this context by blocking on async. SDK call sites already run on threadpool.
+        byte[] bytes = this.utf8.GetBytes(json);
+        string token = this.payLoadStore.UploadAsync(bytes, CancellationToken.None).GetAwaiter().GetResult();
+        return token;
+    }
+
+    /// <summary>
+    /// Deserializes the JSON string or resolves the token to the original value.
+    /// </summary>
+    /// <param name="data">The JSON string or token.</param>
+    /// <param name="targetType">The type to deserialize to.</param>
+    /// <returns>The deserialized value.</returns>
+    public override object? Deserialize(string? data, Type targetType)
+    {
+        if (data is null)
+        {
+            return null;
+        }
+
+        string toDeserialize = data;
+        if (data.StartsWith(TokenPrefix, StringComparison.Ordinal))
```

Comment:
As I mentioned above, this data converter shouldn't have direct knowledge of the blob payload store. See my recommendation above.